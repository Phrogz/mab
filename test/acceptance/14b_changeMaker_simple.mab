setSlot( 'go', method( x, y, branch, depth,
	setSlot( "indent", "  " *( depth ) )
	p( indent +( "go( #{x}, #{y}, #{branch}, #{depth} )" interpolate ) )
	setSlot( "returnValue", if( x ==( 0 ),
		100,
		if( y ==( 0 ),
			1,
			go( x, y -( 1 ), 'a', depth+( 1 ) ) +( go( x -( 1 ), 0, 'b', depth+( 1 ) ) )
		)
	))
	p( indent +( "...returning #{returnValue}" interpolate ) )
	returnValue
))

p( go( 1, 2, '-', 0 ) )
