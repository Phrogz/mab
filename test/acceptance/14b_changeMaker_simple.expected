go( 1, 2, -, 0 )
  go( 1, 1, a, 1 )
    go( 1, 0, a, 2 )
    ...returning 1
    go( 0, 0, b, 2 )
    ...returning 100
  ...returning 101
  go( 0, 0, b, 1 )
  ...returning 100
...returning 201
201